tests/accessor/accessor_exceptions.hpp

```diff
@@ -0,0 +1,380 @@
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from "
+      "buffer, handler and range. In case, the range exceeds the range of "
+      "buffer in any dimension");
+  SECTION(section_name) {
+    auto construct_acc = [&great_range](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension, sycl::access_mode::read, Target>(
+          data_buf, cgh, great_range);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from "
+      "buffer, handler, range and tag. In case, the range exceeds the range of "
+      "buffer in any dimension.");
+  SECTION(section_name) {
+    auto construct_acc = [&great_range](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension>(data_buf, cgh, great_range,
+                                       sycl::read_only);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to construct accessor from "
+      "buffer, handler, range and tag. In case, when the sum of range and "
+      "offset exceeds the range of buffer in any dimension.");
+  SECTION(section_name) {
+    auto construct_acc = [&default_range, id](
+                             sycl::handler& cgh,
+                             sycl::buffer<DataT, Dimension> data_buf) {
+      sycl::accessor<DataT, Dimension, sycl::access_mode::read, Target>(
+          data_buf, cgh, default_range, id);
+    };
+    check_exception<AccType, DataT, Dimension, Target>(construct_acc);
+  }
+}
+
+/**
+ * @brief Struct with functor that will be used in type coverage to call all
+ *        verification functions.
+ * @tparam AccT Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @tparam T Current data type
+ * @tparam AccessModeTypeT Access mode type
+ * @tparam TargetTypeT Target type
+ * @tparam DimensionTypeT Dimension type
+ * @param type_name Current data type string representation
+ * @param access_mode_name Current access mode string representation
+ * @param target_name Current target string representation
+ */
+template <typename AccT, typename T, typename AccessModeTypeT,
+          typename TargetTypeT, typename DimensionTypeT>
+class run_tests {
+  static constexpr sycl::access_mode AccessModeT = AccessModeTypeT::value;
```

Comment:
In this place you getting `sycl::access_mode`, but I can't find where are you using it. Also there are only `read_only` tags in accessors arguments in tests. Did you planned to extend `sycl::access_mode` coverage?

Reply:
Thanks for this, I've done this in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/329/commits/12cce46ee534c9d064d07d0594a1f24ddec7b45a